File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            m.put(nums[i], i);
        }
        int[] res = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int key = target-nums[i];
            if (m.containsKey(key) && i!=m.get(key)) {
                return new int[]{i, m.get(key)};
            }
        }
        return null;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res (2);
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            int find = target - nums[i];
            if (map.find(find) != map.end()) {
                res[0] = i;
                res[1] = map[find];
                break;
            } else {
                map.insert({nums[i], i});
            }
        }
        
        return res;
    }
};